lib/panoramix/query.ex

```diff
@@ -436,19 +442,40 @@ defmodule Panoramix.Query do
     end
   end
 
-  # TODO: handle dimension specs + extraction functions, not just "plain" dimensions
-  defp maybe_build_dimension({{:., _, [{:dimensions, _, _}, dimension]}, _, _}) do
+  # TODO: handle more extraction functions
+  defp dimension_or_extraction_fn({{:., _, [{:dimensions, _, _}, dimension]}, _, _}) do
     # dimensions.foo
-    Atom.to_string dimension
+    %{dimension: Atom.to_string(dimension)}
   end
-  defp maybe_build_dimension({{:., _, [Access, :get]}, _, [{:dimensions, _, _}, dimension]}) do
+  defp dimension_or_extraction_fn({{:., _, [Access, :get]}, _, [{:dimensions, _, _}, dimension]}) do
     # dimensions["foo"]
-    dimension
+    %{dimension: dimension}
   end
-  defp maybe_build_dimension(_) do
+  defp dimension_or_extraction_fn({:lookup, _, args}) do
+    case args do
+      [from_dimension, lookup_name] ->
+        %{dimension: Atom.to_string(from_dimension),
+          extractionFn: {:%{}, [],
+                         [{"type", "registeredLookup"},
+                          {"lookup", lookup_name}]}}
```

Comment:
Not sure how critical it is for a typical use case, but it looks like there's no way to pass a property for registered lookup (e.g. `retainMissingValue`, `replaceMissingValueWith`, `injective`, `optimize`). 
Perhaps, that can be done by allowing a list of properties to be appended in the `args`. E.g.

```
    case args do
      [from_dimension, lookup_name, opts] ->
        %{dimension: Atom.to_string(from_dimension),
          extractionFn: {:%{}, [],
                         [{"type", "registeredLookup"},
                          {"lookup", lookup_name}] ++ opts}}
    ...
```

Reply:
That's a good point, I'll add that.